File: personalized-learning-hub-backend/src/main/java/com/personalized_learning_hub/service/user/SessionReservationService.java
package com.personalized_learning_hub.service.user;

import com.personalized_learning_hub.dto.session.DtoSessionReservationRequest;
import com.personalized_learning_hub.dto.session.DtoSessionReservationResponse;

import java.util.List;

public interface SessionReservationService {


    // ========== CRUD İŞLEMLERİ ==========


    public DtoSessionReservationResponse createReservation(DtoSessionReservationRequest request);
    void deleteReservation(Long reservationId);
    DtoSessionReservationResponse updateReservation(Long reservationId, DtoSessionReservationRequest request);



    // ----------- TUTOR İŞLEMLERİ ------------

    List<DtoSessionReservationResponse> getReservationsByTutor(Long tutorId);



    // =========== STUDENT İŞLEMLERİ =============
    List<DtoSessionReservationResponse> getReservationsByStudent(Long studentId);


}

File: personalized_learning_hub_frontend_react/src/services/sessionReservationService.js
import api from "../utils/axiosConfig";



// =======================
// === CRUD İŞLEMLERİ ===
// =======================

/** Yeni rezervasyon oluşturur */
export const createReservation = async (requestBody) => {
  const response = await api.post(`/reservations`, requestBody);
  return response.data;
};

/** Rezervasyonu günceller */
export const updateReservation = async (reservationId, data) => {
  const response = await api.put(`/reservations/${reservationId}`, data);
  return response.data;
};

/** Rezervasyonu siler */
export const deleteReservation = async (reservationId) => {
  await api.delete(`/reservations/${reservationId}`);
};



// ==========================
// === STUDENT İŞLEMLERİ ====
// ==========================

/** Belirli öğrenciye ait tüm rezervasyonları getirir */
export const getStudentReservations = async (studentId) => {
  const response = await api.get(`/reservations/student/${studentId}`);
  return response.data;
};



// ========================
// === TUTOR İŞLEMLERİ ====
// ========================

/** Belirli eğitmene ait tüm rezervasyonları getirir */
export const getTutorReservations = async (tutorId) => {
  const response = await api.get(`/reservations/tutor/${tutorId}`);
  return response.data;
};
